src/range/bsearch.cr

```diff
@@ -67,6 +67,16 @@
   end
 {% end %}
 
+private def bsearch_increment(x)
+  if x < -1
+    x >> 1
+  elsif x >= -1 && x < 1
+    1
+  else
+    x << 1
```

Comment:
I don't think this is correct since it wraps back to negative numbers without overflowing when `x` is a primitive integer. For example, if I do:

```crystal
(0..).bsearch do |x|
  p x
  x > 0x7FFF0000
end
```

then it prints `1073741824` followed by `-2147483648`, and there is an infinite loop even though there exists an element satifsying the predicate (`0x7FFF0001_i32`).

This makes me wonder if `(x..).bsearch` should always be equivalent to `(x..B::MAX).bsearch` instead. If this is the case, maybe requiring the caller to specify an explicit upper bound isn't so bad after all?